src/session_manager.rs

```diff
@@ -0,0 +1,1056 @@
+//! Widget for managing sessions.
+//!
+//! This widget can be considered as the main view. It is directly subordinate to the application
+//! window and takes care of managing sessions. In the following sections it is described what this
+//! includes.
+//!
+//! # Adding new sessions
+//! The `SessionManager` directs it to [`Login`](struct@crate::login::Login) as soon as the user
+//! wants to add a new session within the function `add_new_session`. Login returns control to the
+//! session manager if either the new session is ready, logging into the new session is aborted by
+//! the user, or it is noticed that the phone number is already logged in (QR code session
+//! duplicates can be detected earliest if the phone number was determined using `fetch_me`).
+//!
+//! # Adding existing sessions
+//! The `SessionManager` analyzes the individual database directories in the Telegrand data
+//! directory to see which sessions can be logged in directly using
+//! [`SessionManager::add_existing_session()`]. To do this, it checks the presence of a `td.binlog`
+//! or a `td_test.binlog` file.
+//!
+//! # Tearing down sessions
+//! This is done by logging the client out and by deleting the database directory. See
+//! (see [`teardown_database()`]) for details.
+//! Tearing down sessions happens in different places: When the login is canceled, When the QR code
+//! is canceled, when a logged in session is logged out, and for all sessions that are not logged
+//! in when the application exits.
+//!
+//! # Migration strategy
+//! There is no special strategy. The legacy database should keep working as before.
```

Comment:
Well, then the section is not really needed :)
Also, the 'legacy' part is still referring to an in-development state, so it's not really useful having that section either way.

Reply:
Thanks! I also removed these sentences from the commit messages.